### DISCUSSION TITLE
Workaround for Nginx Proxy Manager: 'certbot: error: unrecognized arguments' even though I ran '/app/scripts/install-certbot-plugins'

### DISCUSSION THREAD

**Discussion by githubaff0** [community-scripts/ProxmoxVE#831] [General]
Proxmox Virtual Environment 8.3.1
Nginx Proxy Manager v2.12.1 © 2024 jc21.com / Theme by Tabler / Proxmox VE Helper-Scripts.

I just installed NGINX Proxy Manager LXC and ran into an issue with `certbot: error: unrecognized arguments` even though I installed the plugins. I wanted to share the info I found and a potential workaround for anyone else running into the same issue.

 * I installed Nginx Proxy Manager LXC via helper script https://community-scripts.github.io/ProxmoxVE/scripts?id=nginxproxymanager
 * I ran `/app/scripts/install-certbot-plugins` in the nginxproxymanager LXC console to install the plugins
 * In the Nginx Proxy Manager web GUI, I tried to add a Let's Encrypt Certificate using DNS Challenge for the DuckDNS provider
 * I got an error back: `certbot: error: unrecognized arguments: --dns-duckdns-credentials ... --dns-duckdns-no-txt-restore ...`

As noted above, I did run `/app/scripts/install-certbot-plugins` and one of the plugins had an error, but the duckdns plugin seemed to install fine. I reran the `install-certbot-plugins` script again just in case, but still got the certbot error when trying to create the SSL cert.

The issue is something about python dist-packages vs site-packages as reported in this comment in a Nginx Proxy Manager issue: https://github.com/NginxProxyManager/nginx-proxy-manager/issues/4191#issuecomment-2523699811
 * 3 people in that thread that are having the same issue are using ProxmoxVE script to install Nginx Proxy Manager LXC.
 * `/app/scripts/install-certbot-plugins` installs the plugins using a certbot Python virtualenv, e.g. '/opt/certbot/bin/activate && pip install --no-cache-dir  certbot-dns-duckdns~=1.0 && /opt/certbot/bin/deactivate' so they get installed in '/opt/certbot/lib/python3.11/site-packages/'
 * However, when I try to create a cert from the Nginx Proxy Manager Web GUI, it seems the certbot command doesn't see the plugins installed in site-packages (because its not using the virtualenv?).
 * ***The people in the thread said to use the MultiDns plugin instead***, but I wasn't sure how to use MultiDns with DuckDNS.

***The workaround for me was to install the duckdns certbot plugin without the python virtualenv used by /app/scripts/install-certbot-plugins***, i.e. I ran `pip install --no-cache-dir  certbot-dns-duckdns~=1.0` in the nginxproxymanager LXC console, which installed the plugin in /usr/local/lib/python3.11/dist-packages/.

After doing that, I was able to create my SSL cert via the Nginx Proxy Manager Web GUI successfully.

I don't think that's the proper solution for the issue, but I wanted to post the workaround to help anyone else.

I'm out of my depth with Python, but there's an explanation here about Debian using dist-packages vs site-packages: https://stackoverflow.com/questions/9387928/whats-the-difference-between-dist-packages-and-site-packages.

I'm not sure if this a bug in NginxProxyManager or if this is due to some subtle difference between NginxProxyManager under Docker vs NginxProxyManager under ProxmoxVE/LXC or if its a one-off issue for me because of how I installed things because I'm a newbie to Proxmox and the helper scripts.

**Comment by cmayr123** [comment]
Awesome. I had the same issue and your command helped. Thanks!

**Reply by githubaff0** [reply]
Thanks - your feedback confirms it was worth the effort to post the info.

**Comment by metal-x11** [comment]
Can't you just specify to install the duckdns plugin exclusively rather than installing all plugins?

If in root folder:
`/app/scripts/./install-certbot-plugins duckdns`

If in /app/scripts folder:
`./install-certbot-plugins duckdns`

**Reply by metal-x11** [reply]
After testing this on my own, I found I am running into the same issue!
`certbot: error: unrecognized arguments: --dns-duckdns-credentials ... --dns-duckdns-no-txt-restore ...`

**Reply by githubaff0** [reply]
The problem wasn't about installing 1 plugin vs all of them.

It was with how the `/app/scripts/install-certbot-plugins` script installs the plugins. Running the `/app/scripts/install-certbot-plugins` script for a single plugin still puts the plugin in the certbot `site-packages` directory where they don't get seen when the certbot command is run from the GUI.

The workaround is to run the raw `pip` install command directly, i.e. so it runs without the python virtualenv 'wrapper' commands that the `/app/scripts/install-certbot-plugins` script was using around the `pip` install command. This will end up installing the plugin in the (global?) `dist-packages` directory.

A better solution might be to figure out how to get the certbot command to recognize the 'site-packages' directory when the certbot command is run from the GUI, but that will need someone smarter than me to figure out.

**Comment by vsa-d** [comment]
Thanks for sharing.

I am new in nginx. I am using it with since 2 weeks as my reverse-prox-server on proxmox (built with proxmox-helper tteck script).
After setting up my request for a Letsencrypt cert via GUI was successful. (Without installing the /app/scripts/install-certbot-plugins scripts)

Yesterday I finally installed the addon-scripts, I also had some issues with not all addon could installed.

My question (as a beginner) is: How do I enable in nginx to renew my certs from Letsencrpt when its time? (Sorry for this beginner question....)
